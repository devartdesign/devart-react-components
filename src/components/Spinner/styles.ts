import { StyleSheet } from 'aphrodite/no-important';

const spinnerAnimation = {
  from: {
    transform: 'rotate(0)'
  },
  to: {
    transform: 'rotate(359deg)'
  }
};

export default StyleSheet.create({
  container: {
    display: 'flex',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center'
  },
  spinner: {
    display: 'inline-block',
    verticalAlign: 'text-bottom',
    borderStyle: 'solid',
    borderRadius: '50%',
    animationName: spinnerAnimation,
    animationDuration: '1s',
    animationIterationCount: 'infinite',
    animationTimingFunction: 'linear'
  }
});
