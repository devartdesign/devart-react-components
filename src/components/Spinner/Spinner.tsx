import React, { memo } from 'react';
import { css } from 'aphrodite/no-important';

import styles from './styles';

export interface ISpinnerProps {
  size?: number;
  color?: string;
}

const Spinner = ({ size = 25, color = '#333333' }: ISpinnerProps) => {
  return (
    <div key="Spinner" className={css(styles.container)}>
      <div
        className={css(styles.spinner)}
        style={{ width: size, height: size, borderWidth: Math.min(Math.round(size / 6), 5), borderColor: color, borderRightColor: 'transparent' }}
      />
    </div>
  );
};

export default memo(Spinner);
