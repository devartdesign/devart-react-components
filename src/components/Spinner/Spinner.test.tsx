import * as React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import Spinner from './Spinner';
describe('Spinner', () => {
  let Component;
  it('should render', () => {
    Component = shallow<Spinner>(<Spinner />);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });
});
