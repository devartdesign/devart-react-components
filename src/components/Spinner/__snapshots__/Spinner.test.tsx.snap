// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render 1`] = `
.container {
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  height: 100%;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  align-items: center;
}

.spinner {
  display: inline-block;
  vertical-align: text-bottom;
  border-style: solid;
  border-radius: 50%;
  -webkit-animation-name: keyframe_1vzreuq;
  animation-name: keyframe_1vzreuq;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

<div
  className="container"
>
  <div
    className="spinner"
    style={
      Object {
        "borderColor": "#333333",
        "borderRightColor": "transparent",
        "borderWidth": 4,
        "height": 25,
        "width": 25,
      }
    }
  />
</div>
`;
